src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -16,6 +16,291 @@ public enum UpdateStatus
     public class Utils
     {
         /// <summary>
+        /// Builds the dependencies within the list of graphNodes
+        /// </summary>
+        /// <param name="graphNodeScopeToCheck"></param>
+        public static void BuildGraphNodeDependencies(List<AssociativeGraph.GraphNode> graphNodesInScope)
```

Comment:
I think they could be moved to `AssociativeGraph` namespace, instead of being a util class...There always are so many util classes in the VM.